llvm/test/Analysis/LoopAccessAnalysis/depend_diff_types.ll

```diff
@@ -140,11 +140,11 @@ define void @neg_dist_dep_type_size_equivalence(ptr nocapture %vec, i64 %n) {
 ; CHECK-NEXT:            %ld.i64 = load i64, ptr %gep.iv, align 8 ->
 ; CHECK-NEXT:            store i32 %ld.i64.i32, ptr %gep.iv.n.i64, align 8
 ; CHECK-EMPTY:
-; CHECK-NEXT:        BackwardVectorizableButPreventsForwarding:
+; CHECK-NEXT:        BackwardVectorizable:
```

Comment:
Those are all undesired functional changes that don't seem directly related to the goal of the patch?

Reply:
This is unfortunate but true. The problem is that the compiler could decide during loop access analysis, that if distance is not power of 2, the loop cannot be vectorized. With this patch, we cannot do this anymore, we do not know how the loop is going to be vectorized, using power-of-2 fixed vector length or scalable vectorization with non-power-of-2 support.